docs/build.jl

```diff
@@ -16,15 +27,19 @@ cd(dirname(@__FILE__)) do
     end
 
     # Generate and save the contents of docstrings as markdown files.
+    # Example of customary STYLE for API Index
+    plmain = startgenidx(joinpath(api_directory, "genindex.md"); headerstyle = "#", modnamestyle = "##")
     for m in modules
         filename = joinpath(api_directory, "$(module_name(m)).md")
         try
-            save(filename, m)
+            save(filename, m, plmain)
         catch err
             println(err)
             exit(1)
         end
     end
+    # To actually save the API Index to: `genindex.md`: If not index page is needed this can be skipped
+    savegenidx(plmain)
```

Comment:
Could the API be streamlined a bit here. Pass in an index object which is then mutated and then saving the result seems too complex to me.

How about having `save` return an index object and them applying `save` the to resulting indices vector?

Kind of like:

```julia
indices = []
for m in modules
    filename = joinpath(api_directory, "$(module_name(m)).md")
    push!(indices, save(filename, m))
end
save(joinpath(api_directory, "genindex.md"), indices; headerstyle = "#", modnamestyle = "##")
```

I've not investigated whether that's feasible yet, so please do say if it isn't.